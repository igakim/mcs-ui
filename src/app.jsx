import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import App from './view/Home';

export default (store) => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
    document.getElementById('root'),
  );
};
