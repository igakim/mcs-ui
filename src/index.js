import '@babel/polyfill';

// Import Components:
import Button from './components/button/Button';
import Menu from './components/menu/Menu';
import Section from './components/section/Section';
import Card from './components/card/Card';
import Form from './components/form/Form';

export {
  Button,
  Menu,
  Section,
  Card,
  Form,
};
