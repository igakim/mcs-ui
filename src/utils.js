export const isEqualTypes = (type, compareType) => (
  type && type === compareType
);

export const getButtonType = type => (
  type && `mcs-button-${type}`
);

export const getClassNameFromArray = classesList => classesList
  .filter(el => (el !== undefined && el !== null))
  .join('-');

export const isClassComponent = component => (!!(
  typeof component === 'function'
    && component.prototype
    && component.prototype.isReactComponent
));

export const isFunctionComponent = component => (!!((
  typeof component === 'function'
    && !(
      component.prototype
      && component.prototype.isReactComponent
    )
)));

export const isReactComponent = component => (!!((
  isClassComponent(component)
    || isFunctionComponent(component)
)));
