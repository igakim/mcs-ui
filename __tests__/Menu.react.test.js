import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Menu } from '../src';

configure({ adapter: new Adapter() });

test('<Menu /> component should match snapshot', () => {
  const { Item, Logo } = Menu;

  const component = renderer.create(
    <Menu>
      <Logo src="/path/to/pic.png" />
      <Item to="/home">Home</Item>
      <Item to="/article">Article</Item>
      <Item to="/blog">Blog</Item>
    </Menu>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('<Menu /> component should match snapshot', () => {
  const { Item, Logo, DropdownItem } = Menu;

  const dropdownItems = [
    { title: 'Project X', key: 'sub1' },
    { title: 'New project', key: 'sub2' },
    { title: 'Huge project', key: 'sub3' },
  ];

  const snapshot = renderer.create(
    <Menu>
      <Logo textColor="#f00">
        Logo Text
      </Logo>
      <Item to="/blog">Blog</Item>
      <DropdownItem subItems={dropdownItems}></DropdownItem>
    </Menu>
  );

  const tree = snapshot.toJSON();

  expect(tree).toMatchSnapshot();

  const component = shallow(
    <DropdownItem subItems={dropdownItems}>DropdownLink</DropdownItem>
  );

  expect(component.exists('.dropdown-link')).toBeTruthy();
  expect(component.exists('.mcs-menu-dropdown')).toBeFalsy();

  component.find('.dropdown-link').simulate('mouseenter');
  expect(component.exists('.mcs-menu-dropdown')).toBeTruthy();
  component.find('.dropdown-link').simulate('mouseleave');
  setTimeout(expect(component.exists('.mcs-menu-dropdown')).toBeFalsy, 400);
  component.find('.dropdown-link').simulate('mouseenter');
  component.find('.mcs-menu-dropdown').simulate('mouseenter');
  setTimeout(expect(component.exists('.mcs-menu-dropdown')).toBeTruthy, 500);
  component.find('.mcs-menu-dropdown').simulate('mouseleave');
  setTimeout(expect(component.exists('.mcs-menu-dropdown')).toBeFalsy, 500);
});