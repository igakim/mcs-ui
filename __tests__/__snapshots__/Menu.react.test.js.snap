// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> component should match snapshot 1`] = `
<div
  className="mcs-menu  "
>
  <div
    className="mcs-menu-logo "
  >
    <a
      className="mcs-menu-link"
      href="/"
      style={
        Object {
          "color": "#fff",
        }
      }
    />
  </div>
  <div
    className="mcs-menu-items"
  >
    <ul
      className="mcs-menu-list"
    >
      <li
        className="mcs-menu-item "
      >
        <a
          className="mcs-menu-link"
          href="/home"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Home
        </a>
      </li>
      <li
        className="mcs-menu-item "
      >
        <a
          className="mcs-menu-link"
          href="/article"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Article
        </a>
      </li>
      <li
        className="mcs-menu-item "
      >
        <a
          className="mcs-menu-link"
          href="/blog"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Blog
        </a>
      </li>
    </ul>
  </div>
  <div
    className="mcs-menu-side"
  />
  <button
    className="mcs-menu-toggler hide"
    type="button"
  >
    - - -
  </button>
</div>
`;

exports[`<Menu /> component should match snapshot 2`] = `
<div
  className="mcs-menu  "
>
  <div
    className="mcs-menu-logo "
  >
    <a
      className="mcs-menu-link"
      href="/"
      style={
        Object {
          "color": "#f00",
        }
      }
    >
      Logo Text
    </a>
  </div>
  <div
    className="mcs-menu-items"
  >
    <ul
      className="mcs-menu-list"
    >
      <li
        className="mcs-menu-item "
      >
        <a
          className="mcs-menu-link"
          href="/blog"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Blog
        </a>
      </li>
      <li
        className="mcs-menu-item dropdown-item"
      >
        <a
          className="mcs-menu-link dropdown-link"
          href="#"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": undefined,
            }
          }
        >
           
          <svg
            fill="currentColor"
            height=".7em"
            size=".7em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 512 512"
            width=".7em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
  <div
    className="mcs-menu-side"
  />
  <button
    className="mcs-menu-toggler hide"
    type="button"
  >
    - - -
  </button>
</div>
`;
